File: go-client-sdk/pkg/models/shared/codatpublicapimodelsdatatrackingcategory.go
package shared

import (
	"time"
)

// CodatPublicAPIModelsDataTrackingCategory
// Details of a category used for tracking transactions
type CodatPublicAPIModelsDataTrackingCategory struct {
	HasChildren        *bool                                                 `json:"hasChildren,omitempty"`
	ID                 *string                                               `json:"id,omitempty"`
	ModifiedDate       *time.Time                                            `json:"modifiedDate,omitempty"`
	Name               *string                                               `json:"name,omitempty"`
	ParentID           *string                                               `json:"parentId,omitempty"`
	SourceModifiedDate *time.Time                                            `json:"sourceModifiedDate,omitempty"`
	Status             *CodatDataContractsDatasetsTrackingCategoryStatusEnum `json:"status,omitempty"`
}

File: typescript-client-sdk/src/sdk/models/shared/codatpublicapimodelsdatatrackingcategory.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";
import { CodatDataContractsDatasetsTrackingCategoryStatusEnum } from "./codatdatacontractsdatasetstrackingcategorystatusenum";
import { Expose, Transform } from "class-transformer";


// CodatPublicApiModelsDataTrackingCategory
/** 
 * Details of a category used for tracking transactions
**/
export class CodatPublicApiModelsDataTrackingCategory extends SpeakeasyBase {
  @SpeakeasyMetadata()
  @Expose({ name: "hasChildren" })
  hasChildren?: boolean;

  @SpeakeasyMetadata()
  @Expose({ name: "id" })
  id?: string;

  @SpeakeasyMetadata()
  @Expose({ name: "modifiedDate" })
  @Transform(({ value }) => new Date(value), { toClassOnly: true })
  modifiedDate?: Date;

  @SpeakeasyMetadata()
  @Expose({ name: "name" })
  name?: string;

  @SpeakeasyMetadata()
  @Expose({ name: "parentId" })
  parentId?: string;

  @SpeakeasyMetadata()
  @Expose({ name: "sourceModifiedDate" })
  @Transform(({ value }) => new Date(value), { toClassOnly: true })
  sourceModifiedDate?: Date;

  @SpeakeasyMetadata()
  @Expose({ name: "status" })
  status?: CodatDataContractsDatasetsTrackingCategoryStatusEnum;
}